django/db/backends/base/base.py

```diff
@@ -75,6 +75,9 @@ def __init__(self, settings_dict, alias=DEFAULT_DB_ALIAS):
         self.needs_rollback = False
 
         # Connection termination related attributes.
+        self.health_check_at = None
```

Comment:
I haven't followed the entire logic, but I think you can minimize the mutable state a bit by instead keeping `health_check_delay` (immutable) and `last_health_check_at: Optional[float]`, where `None` means same as `health_check_done == False` and otherwise `health_check_at` is `last_health_check_at + health_check_delay`.